test/unit/sql/cte.test.js

```diff
@@ -0,0 +1,304 @@
+          });
+      });
+      
+      if (current.dialect.supports.cteLimitOffsetOrder) {
+
+        it('correctly adds order to recursion', function () {
+
+          selectTest(
+            {
+              cte: [{
+                name: 'a',
+                initial: {
+                  where: { username: 'user3' }
+                },
+                recursive: {
+                  next: 'report',
+                },
+                order: [['amount', 'ASC']]
+              }]
+            },
+            {
+              default: 'WITH RECURSIVE a( [id],[amount],[createdAt],[updatedAt],[userId] ) AS ( SELECT [id], [amount], [createdAt], [updatedAt], [userId] FROM [users] AS [user] WHERE [user].[username] = \'user3\' UNION SELECT [report].[id], [report].[amount], [report].[createdAt], [report].[updatedAt], [report].[userId] FROM [a] INNER JOIN [users] AS [report] ON [a].[id] = [report].[userId]  ORDER BY 2 ASC ) SELECT [user].* FROM [users] AS [user] INNER JOIN [a] ON [user].[id] = [a].[id];'
+            });
+        });
+
+        it('correctly adds limit & offset to recursion', function () {
+
+          selectTest(
+            {
+              cte: [{
+                name: 'a',
+                initial: {
+                  where: { username: 'user3' }
+                },
+                recursive: {
+                  next: 'report',
+                },
+                limit: 2,
+                offset: 2
+              }]
+            },
+            {
+              default: 'WITH RECURSIVE a( [id],[amount],[createdAt],[updatedAt],[userId] ) AS ( SELECT [id], [amount], [createdAt], [updatedAt], [userId] FROM [users] AS [user] WHERE [user].[username] = \'user3\' UNION SELECT [report].[id], [report].[amount], [report].[createdAt], [report].[updatedAt], [report].[userId] FROM [a] INNER JOIN [users] AS [report] ON [a].[id] = [report].[userId]  LIMIT 2, 2 ) SELECT [user].* FROM [users] AS [user] INNER JOIN [a] ON [user].[id] = [a].[id];'
+            });
+        });
+      
+      
+        it('correctly adds order to CTE with no recursion', function () {
+
+          selectTest(
+            {
+              cte: [{
+                name: 'a',
+                initial: {
+                  where: { username: 'user3' }
+                },
+                order: [['amount', 'ASC']],
+              }]
+            },
+            {
+              default: 'WITH RECURSIVE a( [id],[amount],[createdAt],[updatedAt],[userId] ) AS ( SELECT [id], [amount], [createdAt], [updatedAt], [userId] FROM [users] AS [user] WHERE [user].[username] = \'user3\'  ORDER BY 2 ASC ) SELECT [user].* FROM [users] AS [user] INNER JOIN [a] ON [user].[id] = [a].[id];'
```

Comment:
I recalled reading that the `RECURSIVE` didn't actually matter if the statement wasn't recursive, and it turns out that was from the sqlite docs, which clearly does not speak for all possible dialects. I will go ahead and make the `RECURSIVE` statement conditional on there being a `recursive` object in the find options.